test/integration/compose/application-manager.spec.ts

```diff
@@ -1417,4 +1417,349 @@ describe('compose/application-manager', () => {
 			});
 		});
 	});
+
+	// In the case where a container requires a host resource such as a network interface that is not created by the time the Engine
+	// comes up, the Engine will not attempt to restart the container which seems to be Docker's implemented behavior (if not the correct behavior).
+	// An example of a host resource would be a port binding such as 192.168.88.1:3000:3000, where the IP is an interface delayed in creation by host.
+	// In this case, the Supervisor needs to wait a grace period for the Engine to start the container, and if this does not occur, the Supervisor
+	// deduces the existence of this race condition and generates another start step after a delay (1 minute by default).
+	describe('handling Engine restart policy inaction when host resource required by container is delayed in creation', () => {
+		// Time 61 seconds ago
+		const date61SecondsAgo = new Date();
+		date61SecondsAgo.setSeconds(date61SecondsAgo.getSeconds() - 61);
+
+		// Time 59 seconds ago
+		const date50SecondsAgo = new Date();
+		date50SecondsAgo.setSeconds(date50SecondsAgo.getSeconds() - 50);
+
+		// TODO: We need to be able to start a service with restart policy "no" if that service did not start at all due to
+		// the host resource race condition described above. However, this is harder to parse as the containers do not include
+		// the proper metadata for this. The last resort would be parsing the error message that caused the container to exit.
```

Comment:
This is an outdated comment as the case of restart policy 'no' is already handled by the state funnel without any changes added by this PR. If a service with policy 'no' has never been started, its status is `Installed` which the Supervisor will issue a start step for. Comment to be removed in a future PR.